clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2675,7 +2678,14 @@ protected void setupInputFormatConf(JobConf jobConf, PushJobSchemaInfo pushJobSc
   }
 
   private void setupReducerConf(JobConf jobConf, PushJobSetting pushJobSetting, TopicInfo topicInfo) {
-    jobConf.setPartitionerClass(this.mapRedPartitionerClass);
```

Comment:
Is my understanding correct for the whole process?
1. RecordReader reads all the k-v from Kafka and compose <KafkaInputKey, KafkaInputValue>
2. Mapper seems to do nothing but passing it through
3. During the shuffle phase, combine will group values of the same key by key bytes only, and then sort value again by offset.

but I don't get why we set setOutputValueGroupingComaparator / setCombinerKeyGroupingComparator as the same ?

Reply:
Synced offline and get to understand how this works. Will close this for now and do another pass.